internal/devbox/util.go

```diff
@@ -32,14 +35,66 @@ func (d *Devbox) addDevboxUtilityPackage(ctx context.Context, pkgName string) er
 		return err
 	}
 
-	return nixprofile.ProfileInstall(ctx, &nixprofile.ProfileInstallArgs{
+	return nix.ProfileInstall(ctx, &nix.ProfileInstallArgs{
 		Installable: installable,
-		PackageName: pkgName,
 		ProfilePath: profilePath,
 		Writer:      d.stderr,
 	})
 }
 
+func (d *Devbox) removeDevboxUtilityPackage(pkgName string) error {
+	pkg := devpkg.PackageFromStringWithDefaults(pkgName, d.lockfile)
+	installable, err := pkg.Installable()
+	if err != nil {
+		return err
+	}
+
+	utilProfile := nix.NixProfile{}
+	utilityProfilePath, err := utilityNixProfilePath()
+	if err != nil {
+		return err
+	}
+	profileString, err := nix.ProfileList(d.stderr, utilityProfilePath, true)
+	if err != nil {
+		return err
+	}
+
+	if err = json.Unmarshal([]byte(profileString), &utilProfile); err != nil {
+		return err
+	}
```

Comment:
You should use `nixprofile.ProfileListItems` which handles legacy nix outputs. It may also simplify finding the index.

Edit: actually, you can use `ProfileListNameOrIndex` to find the index.

Reply:
I'm not really sure why we have nixprofile and nix.profile in the source code, but both of those methods you mentioned seem to require that we have a lockfile, which does not exist in the case of the utility profile (since we're manipulating the profile directly, instead of using Devbox). 

I think we'd need to either refactor those methods to not require a profile, or refactor the utility profile to use Devbox.

Reply:
Went with using ProfileListItems to get the list, and then doing a quick match based on unlockedReference == installable. This should work for utility packages that we install from a Flake reference and not from a store path, which is currently the case for all utility packages.